torch/ao/quantization/fx/utils.py

```diff
@@ -455,6 +456,62 @@ def all_node_args_have_no_tensors(node: Node, modules: Dict[str, torch.nn.Module
         cache[node] = result
     return result
 
+def all_node_args_except_first(node: Node) -> List[int]:
+    """
+    Returns all node arg indexes after first
+    """
+    return [i for i, arg in enumerate(node.args)][1:]
+
+NodeInfo = namedtuple("NodeInfo", "op target")
+
+# this dict identifies which indices of a node are non tensors
+# so that they can be propagated correctly since inserting observers
+# for them would cause errors
+
+NON_TENSOR_ARG_DICT = {
+    NodeInfo("call_method", "masked_fill") : {
+        torch.bool: [1],
```

Comment:
this is a tensor arg with dtype boolean, should we rename the data structure to communicate that it can contain indices of both tensors and non tensors?

Reply:
good idea